unpack-trees.c

```diff
@@ -1437,7 +1437,9 @@ static void mark_new_skip_worktree(struct exclude_list *el,
 	 * 2. Widen worktree according to sparse-checkout file.
 	 * Matched entries will have skip_wt_flag cleared (i.e. "in")
 	 */
+	enable_fscache(0);
 	clear_ce_flags(istate, select_flag, skip_wt_flag, el);
+	disable_fscache();
```

Comment:
You may want to mention in the commit message that `disable_fscache()` decrements a counter, and only actually discards the FSCache when that counter reaches 0. (To address one of @jeffhostetler's concerns.)